src/scope/core/pipelines/base/vae/krea_realtime_video.py

```diff
@@ -0,0 +1,209 @@
+"""KreaRealtimeVideo VAE wrapper with encoder/decoder separation."""
+
+import os
+
+import torch
+import torch.nn as nn
+
+from ...krea_realtime_video.modules.vae import CausalConv3d, WanVAE
+from ...krea_realtime_video.modules.vae_block3 import VAEDecoder3d
+
+
+class KreaRealtimeVideoVAE(torch.nn.Module):
+    """VAE wrapper for KreaRealtimeVideo with separated encoder/decoder.
+
+    This VAE uses a specialized architecture optimized for real-time video
+    generation with cached encoding and decoding.
+    """
+
+    def __init__(
+        self,
+        model_name: str = "Wan2.1-T2V-1.3B",
+        model_dir: str | None = None,
+        vae_path: str | None = None,
+    ):
+        super().__init__()
+
+        # Determine paths with priority: specific paths > model_dir > default
+        if vae_path is None:
+            model_dir = model_dir if model_dir is not None else "wan_models"
+            model_path = os.path.join(model_dir, model_name)
+            vae_path = os.path.join(model_path, "Wan2.1_VAE.pth")
+
+        vae = WanVAE(vae_pth=vae_path)
+
+        self.encoder = VAEEncoderWrapper(vae)
+        self.encoder.eval()
+        self.encoder.requires_grad_(False)
+
+        self.decoder = VAEDecoderWrapper()
+        state_dict = torch.load(vae_path, map_location="cpu")
+        decoder_state_dict = {}
+        for key, value in state_dict.items():
+            if "decoder." in key or "conv2" in key:
+                decoder_state_dict[key] = value
```

Comment:
What is this code block doing? The variable `decoder_state_dict` does not seem to be used anywhere.